### PR TITLE
updated  plugin docs

### PR DESCRIPTION
**Pull request by bcoca**
##### SUMMARY
- for devs:
  - added inventory/vars section
  - made some updates to general section and other plugin types
- for users:
 - added 'user' plugin section to start describing the plugins
 - docs on types, what they are and how to use

TODO: link to docs generated from 'plugin embeded docs'

##### ISSUE TYPE
 - Docs Pull Request

##### COMPONENT NAME

docs
##### ANSIBLE VERSION

```
2.4/2.5
```

### CODE REVIEW

**Review by dharmabumstead — APPROVED** [review]
Approved with copy edits. Thanks @bcoca!

**Review by dharmabumstead — CHANGES_REQUESTED** [review]
Copy edit pass in progress. Please do not merge until the docs team signs off.

File: docs/docsite/rst/plugins/shell.rst [hunk 1]
```diff
@@ -0,0 +1,30 @@
+Shell Plugins
+-------------
+
+Shell plugins work behind the scenes making sure the basic commands Ansible runs in order to be able to execute a task's action are
+properly formated to work with the target machine. You normally don't have to wory about these plugins at all unless you have a restricted or exotic setup in which the default ``/bin/sh`` is not a POSIX compatible shell or not available for execution.
+
+Enabling Shell Plugins
+++++++++++++++++++++++
+
+Aside from the default configuration settings in :doc:`../config`, you can use a 'connection variable'
```
Comment (dharmabumstead) [inline docs/docsite/rst/plugins/shell.rst]: Is this correct or copy-pasta?

**Review by dharmabumstead — CHANGES_REQUESTED** [review]
Approved with copy edits, except for one review comment.

File: docs/docsite/rst/plugins/shell.rst (hunk 1 above)
Comment (bcoca) [reply docs/docsite/rst/plugins/shell.rst]: yes

Comment (dharmabumstead) [reply docs/docsite/rst/plugins/shell.rst]: Which is it?

Comment (bcoca) [reply docs/docsite/rst/plugins/shell.rst]: it is correct

**Pull request merged**